python_modules/libraries/dagster-docker/dagster_docker/external_resource.py

```diff
@@ -69,55 +84,64 @@ def run(
                     password=registry["password"],
                 )
 
-            # will need to deal with when its necessary to pull the image before starting the container
-            # client.images.pull(image)
-
-            container = client.containers.create(
-                image=image,
-                command=command,
-                detach=True,
-                environment={**self.get_base_env(), **(self.env or {}), **(env or {}), **io_env},
-                volumes={
-                    **(volumes or {}),
-                    **io_volumes,
-                },
-            )
+            try:
+                container = self._create_container(
+                    client, image, command, ext_env, container_kwargs
+                )
+            except docker.errors.ImageNotFound:
+                client.images.pull(image)
+                container = self._create_container(
+                    client, image, command, ext_env, container_kwargs
+                )
```

Comment:
what if `images.pull` fails?

Reply:
exception bubbles and the ext run / op fails

Reply:
pattern lifted from https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-docker/dagster_docker/docker_run_launcher.py
/ https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-docker/dagster_docker/ops/docker_container_op.py

Reply:
🫡